invisible_cities/cities/detsim.py

```diff
@@ -0,0 +1,216 @@
+from .. detsim.simulate_electrons import generate_ionization_electrons
+from .. detsim.simulate_electrons import drift_electrons
+from .. detsim.simulate_electrons import diffuse_electrons
+from .. detsim.light_tables_c     import LT_SiPM
+from .. detsim.light_tables_c     import LT_PMT
+from .. detsim.s2_waveforms_c     import create_wfs
+from .. detsim.detsim_waveforms   import s1_waveforms_creator
+
+
+def hits_selector (active_only: bool=True):
+    """
+    Filtering function that selects hits. (see :active_only: description)
+
+    Parameters:
+        :active_only: bool
+            if True, returns hits in ACTIVE
+            if False, returns hits in ACTIVE and BUFFER
+    Returns:
+        :select_hits: Callable
+            function that select the hits depending on :active_only: parameter
+    """
+    def select_hits(x, y, z, energy, time, label):
+        sel = (label == "ACTIVE")
+        if not active_only:
+            sel =  sel | (label == "BUFFER")
+        return x[sel], y[sel], z[sel], energy[sel], time[sel], label[sel]
+    return select_hits
+
+
+def ielectron_simulator(wi: float, fano_factor: float, lifetime: float,
+                        transverse_diffusion: float, longitudinal_diffusion: float, drift_velocity:float,
+                        el_gain: float, el_gain_sigma: float):
+    """
+    Function that simulates electron creation, drift, diffusion and photon generation at the EL
+
+    Parameters: floats
+        parameter names are self-descriptive.
+    Returns:
+        :simulate_ielectrons:
+            function that returns the positions emission times and number of photons at the EL
+    """
+    def simulate_ielectrons(x, y, z, time, energy):
+        electrons = generate_ionization_electrons(energy, wi, fano_factor)
+        electrons = drift_electrons(z, electrons, lifetime, drift_velocity)
+        dx, dy, dz = diffuse_electrons(x, y, z, electrons, transverse_diffusion, longitudinal_diffusion)
+        dtimes = dz/drift_velocity + np.repeat(time, electrons)
+        nphotons = np.round(np.random.normal(el_gain, el_gain_sigma, size=electrons.sum())).astype(np.int32)
+        return dx, dy, dz, dtimes, nphotons
+    return simulate_ielectrons
+
+
+def buffer_times_and_length_getter(wf_pmt_bin_width, wf_sipm_bin_width, el_gap, el_dv, max_length):
+    """
+    Auxiliar function that computes the signal absolute starting-time and an estimated buffer_length
+    """
+    max_sensor_bin = max(wf_pmt_bin_width, wf_sipm_bin_width)
+    def get_buffer_times_and_length(time, times_ph):
+        start_time = np.floor(min(time) / max_sensor_bin) * max_sensor_bin
+        el_traverse_time = el_gap / el_dv
+        end_time   = np.ceil((max(times_ph) + el_traverse_time)/max_sensor_bin) * max_sensor_bin
+        buffer_length = min(max_length, end_time-start_time)
+        return start_time, buffer_length
+    return get_buffer_times_and_length
+
+
+def s2_waveform_creator (sns_bin_width, LT, EL_drift_velocity):
```

Comment:
use lowercase for variables (uppercase for classes instances are ok)

Reply:
If we are used to seeing it upper case in our standard notation outside of source code, I would recommend sticking to uppercase even in source code.

But we do follow PEP-8 on *no-spaces-before-parentheses*, unless there's an overriding reason (usually vertical alignment) to do otherwise: I don't see such a reason here.